src/wms/src/main/java/org/geoserver/wms/legendgraphic/LegendMerger.java

```diff
@@ -0,0 +1,795 @@
+            this.forceLabelsOn = forceLabelsOn;
+            this.forceLabelsOff = forceLabelsOff;
+        }
+        public List<RenderedImage> getImageStack() {
+            return imageStack;
+        }
+        public void setImageStack(List<RenderedImage> imageStack) {
+            this.imageStack = imageStack;
+        }
+        public int getDx() {
+            return dx;
+        }
+        public void setDx(int dx) {
+            this.dx = dx;
+        }
+        public int getDy() {
+            return dy;
+        }
+        public void setDy(int dy) {
+            this.dy = dy;
+        }
+        public int getMargin() {
+            return margin;
+        }
+        public void setMargin(int margin) {
+            this.margin = margin;
+        }
+        public Color getBackgroundColor() {
+            return backgroundColor;
+        }
+        public void setBackgroundColor(Color backgroundColor) {
+            this.backgroundColor = backgroundColor;
+        }
+        public boolean isTransparent() {
+            return transparent;
+        }
+        public void setTransparent(boolean transparent) {
+            this.transparent = transparent;
+        }
+        public boolean isAntialias() {
+            return antialias;
+        }
+        public void setAntialias(boolean antialias) {
+            this.antialias = antialias;
+        }
+        public LegendLayout getLayout() {
+            return layout;
+        }
+        public void setLayout(LegendLayout layout) {
+            this.layout = layout;
+        }
+        public int getRowWidth() {
+            return rowWidth;
+        }
+        public void setRowWidth(int rowWidth) {
+            this.rowWidth = rowWidth;
+        }
+        public int getRows() {
+            return rows;
+        }
+        public void setRows(int rows) {
+            this.rows = rows;
+        }
+        public int getColumnHeight() {
+            return columnHeight;
+        }
+        public void setColumnHeight(int columnHeight) {
+            this.columnHeight = columnHeight;
+        }
+        public int getColumns() {
+            return columns;
+        }
+        public void setColumns(int columns) {
+            this.columns = columns;
+        }
+        public Font getLabelFont() {
+            return labelFont;
+        }
+        public void setLabelFont(Font labelFont) {
+            this.labelFont = labelFont;
+        }
+        public boolean isForceLabelsOn() {
+            return forceLabelsOn;
+        }
+        public void setForceLabelsOn(boolean forceLabelsOn) {
+            this.forceLabelsOn = forceLabelsOn;
+        }
+        public boolean isForceLabelsOff() {
+            return forceLabelsOff;
+        }
+        public void setForceLabelsOff(boolean forceLabelsOff) {
+            this.forceLabelsOff = forceLabelsOff;
+        }
+        public static MergeOptions createFromRequest(List<RenderedImage> imageStack, int dx, int dy, int margin, 

```

Comment:
You may just want to create a constructor with this signature? The constructor can call "this" ...